README.md

```diff
@@ -1,155 +1,195 @@
+key for your MagicBell project is required.
 
-First, you must configure your MagicBell instance. Just create a MagicBell object as follows:
-
-```swift 
-let magicBell = MagicBell(apiKey: "YOUR_API_KEY")
+```swift
+let client = MagicBellClient(apiKey: "[MAGICBELL_API_KEY]")
 ```
 
-Additionally, you might want to add more options to the initialization:
+You can provide additional options when initializing a client:
 
-```swift 
-let magicBell = MagicBell(
-    apiKey: "YOUR_API_KEY"
-    apiSecret: "YOUR_API_SECRET",
-    enableHMAC: true | false,
-    baseUrl: "CUSTOM_MAGICBELL_HOST",
-    logLevel: .none | .debug
+```swift
+let client = MagicBellClient(
+    apiKey: "[MAGICBELL_API_KEY]"
+    apiSecret: "[MAGICBELL_API_SECRET]",
+    enableHMAC: true,
+    logLevel: .debug
 )
 ```
 
+| Param        | Default Value | Description                                                                                  |
+| ------------ | ------------- | -------------------------------------------------------------------------------------------- |
+| `apiKey`     | -             | Your MagicBell's API key                                                                     |
+| `apiSecret`  | `nil`         | Your MagicBell's API secret                                                                  |
+| `enableHMAC` | `false`       | Set it to `true` if you want HMAC enabled. Note the `apiSecret` is required if set to `true` |
+| `logLevel`   | `.none`       | Set it to `.debug` to enable logs                                                            |
 
-| Param | Default Value | Description |
-| - | - | - |
-| `apiKey` | - | Your MagicBell's API key. |
-| `apiSecret` | `nil` | Your MagicBell's API Secret. |
-| `enableHMAC` | `false` | Set to `true` if you want HMAC enabled. Note the `apiSecret` is required if set to `true`.
-| `baseURL` | `https://api.magicbell.com` | MagicBell host. Only customize if running a private instance of MagicBell. |
-| `logLevel` | `.none` | Enables MagicBell logs if set to `.debug`.|
+Though the API key is meant to be published, you should not distribute the API secret. Rather, enable HAMC in your project and generate the user secret on your backend before distributing your app.
 
+### Integrating into your app
 
-## Managing the MagicBell's instance
+You should create the client instance as early as possible in your application and ensure that only one instance is used across your application.
 
-You might want to make your `magicBell` instance accessible from all of your app. 
+To achieve it, extend `MagicBell` and set the instance in your application's delegate (at either the `AppDelegate.swift` or `App.swift` files):
 
-Therefore, you can instantiate it in the `AppDelegate.swift` or `App.swift` file (or any other file) as a constant value. 
+```swift
+import MagicBell
+
+extension MagicBell {
+    static var shared: MagicBellClient!
+}
+```
```

Comment:
Hi @josuemontano,

This is not working on Swift. `MagicBell` is a module and cannot be extended. Still, if it was a class or a struct, you can't extend by adding attributes (you can only add methods).

Therefore, I'd leave it as it was before: a global attribute `let magicBell = MagicBellClient(apiKey: "...")`. It's actually a good approach on Swift, and quite simple to understand.

Reply:
If you want, I can edit the snippet. Let me know.

Reply:
Also, note the whole README.md file uses in its snippets `magicBell` as the client instance. If we change this approach, we should also change all the snippets to use the default way to access the main instance.

Reply:
@vilanovi What would it take to make `MagicBell` a class or struct that can be extended?

Reply:
@vilanovi What would it take to make `MagicBell` a type that can be extended?

Reply:
We did have it, with the `MagicBell` class, and we removed it in favor of uniquely having named `MagicBell` the module.

Still, an extension cannot contain new properties (static or instance-based), therefore a user could not extend it to add a custom `shared` static property.